### PR TITLE
[RORDEV-617] ECK + docker images documentation

### PR DESCRIPTION
**Pull request by coutoPL**

### CODE REVIEW

File: examples/secure_metricbeats.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Secure Metricbeats
```
Comment (Dzuming) [inline examples/secure_metricbeats.md:1]: Shouldn't it be Metricbeat?

image

File: kibana.md [hunk 2]
```diff
@@ -1278,8 +1324,8 @@ Now try to login as user1, and ReadonlyREST Enterprise should initialize the ind
 
 You can configure the `number_of_shards` and `number_of_replicas` for the tenant index via the `kibana.yml` file, allowing you to override the default index settings. This can be particularly useful in a single-node environment.
 ```yaml
-readonlyrest.tenantIndex.number_of_shards: 1
-readonlyrest.tenantIndex.number_of_replicas: 0
+readonlyrest_kbn.tenantIndex.number_of_shards: 1
```
Comment (Dzuming) [inline kibana.md:1327]: This change was on the master branch, i merged it to the develop. Could you update your branch?

**Review by Dzuming — APPROVED** [review]
LGTM

**Comment by sscarduzio** [comment]
> If running POD with root privileges is something you cannot accept, you can create your own image with the patching step done at the image creation level (not in the runtime as our image does) and place it your your registry

then why don't we provide this ourself?

> It's possible, but we don't recommend installing the ReadonlyREST plugin using the init-container pattern, because of the patching required patching step

This needs expansion. How? Why?

File: eck.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
+# ECK 
+
+ReadonlyREST plugins officially support installation on Elasticsearch and Kibana working in K8S cluster and managed by the [ECK operator](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-quickstart.html).
```
Comment (sscarduzio) [inline eck.md:3]: Let's use "Kubernetes", the "k8s" is too informal.

File: eck.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        runAsNonRoot: false
+        runAsUser: 0
+        runAsGroup: 0
+      containers:
+        - name: kibana
+          env:
+            # we have to explicitly agree to patch the KBN binaries (the patching step will be done only once)
+            - name: I_UNDERSTAND_IMPLICATION_OF_KBN_PATCHING
+              value: "yes"
+            # we have to provide a ROR license if we want to use ROR Pro or Enterprise
```
Comment (sscarduzio) [inline eck.md:131]: where do we get the keys? Link to the portal.
can we proceed without keys?

**Comment by coutoPL** [comment]
> > If running POD with root privileges is something you cannot accept, you can create your own image with the patching step done at the image creation level (not in the runtime as our image does) and place it your your registry
> 
> then why don't we provide this ourself?
> 
> > It's possible, but we don't recommend installing the ReadonlyREST plugin using the init-container pattern, because of the patching required patching step
> 
> This needs expansion. How? Why?

Yeah, you are right. See how it looks now

**Comment by coutoPL** [comment]
@sscarduzio WDYT?

File: eck.md [hunk 5]
```diff
@@ -128,19 +137,176 @@ spec:
...
             - name: ROR_ACTIVATION_KEY
               value: "<YOUR_ACTIVATION_KEY/>"
 ```
 
 And these are all differences we need to make to run [ECK Quickstart](https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-quickstart.html) with ReadonlyREST instead of X-Pack security. 
 
+## Using Docker images built by you and stored in your registry
+
+As you probably noticed, our docker images have to be run with root privileges. It's due to legal reasons and you, as a user, 
+have to confirm that you agree to do the patching steps. If running POD with root privileges is something you cannot accept, you can create your own image with the patching step done at the image creation level (not in the runtime as our image does) and place it your your registry.
```
Comment (sscarduzio) [inline eck.md:150]: * running a pod
* not at runtime
* save it on your own registry

File: eck.md [hunk 6]
```diff
@@ -128,19 +137,176 @@ spec:
...
+
+### Elasticsearch with ROR custom image
```
Comment (sscarduzio) [inline eck.md:152]: this can be on its own page?

**Review by sscarduzio — COMMENTED** [review]
Cool! Left some minor coments :)

File: eck.md (hunk 6 above)
Comment (sscarduzio) [reply eck.md:152]: The reason I wanted to isolate this special case into its own space is to keep the basic guide small and easy. People who want their own images bla bla it's something way more involved and I don't want people to get scared of it the first time they land here.

I guess the collapsible can do.

Comment (mk-software-pl) [reply eck.md]: ok, done. Please, take a look

Comment (coutoPL) [reply eck.md]: ok, done. please take a look

Comment (coutoPL) [reply eck.md]: @sscarduzio is it ok now?

**Review by sscarduzio — APPROVED** [review]

**Pull request merged**